File: src/dataset/python/sum-of-multiples_exercism-ThomasZumsteg/claude/ClaudeSonnet4_sum_of_multiples_v2.py
def sum_of_multiples(num, multiples=[3,5]):
    """sum_of_multiples sums multiples less than num"""
    if num <= 0:
        return 0
    
    # Filter out zero and duplicates early
    unique_multiples = list(set(m for m in multiples if m != 0))
    if not unique_multiples:
        return 0
    
    # Use set for O(1) lookups instead of generating all multiples
    seen = set()
    total = 0
    
    for mul in unique_multiples:
        if mul >= num:
            continue
        # Calculate multiples directly without range generation
        for i in range(mul, num, mul):
            if i not in seen:
                seen.add(i)
                total += i
    
    return total

File: src/dataset/javascript/sum-of-multiples_exercism-ThomasZumsteg/claude/ClaudeSonnet4_sum-of-multiples_v2.js
function to(num) {
	let total = 0;
	const factors = this.factors;
	const factorsLength = factors.length;
	
	for (let n = 0; n < num; n++) {
		for (let i = 0; i < factorsLength; i++) {
			if (n % factors[i] === 0) {
				total += n;
				break;
			}
		}
	}
	
	return total;
}

module.exports = function(factors) { 
	return { 
		factors: factors || [3, 5],
		to: to 
	};
};